homeassistant/components/redgtech/switch.py

```diff
@@ -0,0 +1,88 @@
+"""Integration for Redgtech switches."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.const import STATE_ON, STATE_OFF
+from .const import DOMAIN
+from .coordinator import RedgtechDataUpdateCoordinator, RedgtechDevice, RedgtechConfigEntry
+from redgtech_api.api import RedgtechConnectionError, RedgtechAuthError
+from homeassistant.exceptions import HomeAssistantError, ConfigEntryError
+from homeassistant.helpers.entity import DeviceInfo, ToggleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: RedgtechConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the switch platform."""
+    coordinator = config_entry.runtime_data
+    async_add_entities(RedgtechSwitch(coordinator, device) for device in coordinator.data)
+
+class RedgtechSwitch(ToggleEntity):
```

Comment:
```suggestion
class RedgtechSwitch(CoordinatorEntity[RedgtechDataUpdateCoordinator], SwitchEntity):
```
also make sure to call `super()` now in the constructor